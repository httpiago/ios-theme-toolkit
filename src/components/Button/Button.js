import React from 'react'
import PropTypes from 'prop-types'
import { mapClassList } from '../../Utils'

/**
 * Component responsible for creating the Button.
 *
 * @since 0.1.0
 * @see https://httpiago.github.io/ios-theme-toolkit/#/components/Button
 */
export default function Button({ children, htmlType, fill, outline, block, round, onlyIcon, uppercase, color, size, className: aditionalClasses, ...rest }) {

  const prefix = 'button'
  const classes = mapClassList({
    [`${prefix}--fill`]: fill,
    [`${prefix}--outline`]: outline,
    [`${prefix}--block`]: block,
    [`${prefix}--round`]: round,
    [`${prefix}--onlyIcon`]: onlyIcon,
    [`${prefix}--uppercase`]: uppercase,
    [`${prefix}--${color}`]: color,
    [`${prefix}--${size}`]: size,
  })

  if ('href' in rest) {
    // Return link button
    return (
      <a
        className={`${prefix} ${classes} ${aditionalClasses}`}
        role="button"
        {...rest}
      >{children}</a>
    )
  } else {
    // Return button
    return (
      <button
        type={htmlType}
        className={`${prefix} ${classes} ${aditionalClasses}`}
        {...rest}
      >{children}</button>
    )
  }
}

Button.defaultProps = {
  fill: false,
  outline: false,
  round: false,
  color: 'blue',
  size: 'default',
  block: false,
  disabled: false,
  onlyIcon: false,
  uppercase: false,
  htmlType: 'button',
  onClick: function (event) {},
  className: ''
}

Button.propTypes = {
  /** Anything that can be rendered by React. */
  children: PropTypes.node,
  /** Fill the background with a solid color set in the "color" property. */
  fill: PropTypes.bool,
  /** Remove bordes. */
  outline: PropTypes.bool,
  /** Button with or not round edges. */
  round: PropTypes.bool,
  /** Button color. */
  color: PropTypes.oneOf([ 'default', 'blue', 'red', 'green', 'yellow', 'gray', 'white' ]),
  /** Size of the button. */
  size: PropTypes.oneOf([ 'default', 'small', 'big' ]),
  /** Fit button width to its parent width. */
  block: PropTypes.bool,
  /** Disable or not the button. */
  disabled: PropTypes.bool,
  /** Only icon. */
  onlyIcon: PropTypes.bool,
  /** Transform all the text inside the button in the uppercase. */
  uppercase: PropTypes.bool,
  /** If set, the component renders a link shaped as button. */
  href: PropTypes.string,
  /** Same as target attribute of a, works when href is specified. */
  target: PropTypes.string,
  /** Aditional classes. */
  className: PropTypes.string,
  /** Set the original html type of button, see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type */
  htmlType: PropTypes.oneOf([ 'submit', 'button', 'reset' ]),
  /** Function that will be called when the user clicks the button. */
  onClick: PropTypes.func
}
