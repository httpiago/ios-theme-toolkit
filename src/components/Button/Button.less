@import '../../customizations.less';

// Default style
.button {
  background-color: transparent;
  -webkit-appearance: none;
  appearance: none;
  border: 1px solid @gray-color;
  color: @gray-color;
  border-radius: @base-border-radius;
  text-decoration: none;
  text-align: center;
  display: inline-block;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  margin: 0;
  font-size: @font-size;
  line-height: 26px;
  padding: 0 10px;
  height: 30px;
  white-space: nowrap;
  text-overflow: ellipsis;
  position: relative;
  overflow: hidden;
  font-family: @font-family;
  cursor: pointer;
  outline: 0;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.button:hover {

}

// focus
.button:not(:active):focus {
  box-shadow: 0 0 0 3px fade(@gray-color, 50%);
  z-index: 10;
}

// Botão desativado
.button[disabled] {
  background-color: @gray-color;
  border-color: transparent;
  color: #FFF;
  opacity: .7;
  pointer-events: none;
}

// Botão sem bordas
.button--outline {
  border-color: transparent !important;
}

// Botão com preenchimento
.button--fill {
  background-color: @gray-color;
  border-color: transparent;
  color: #FFF;

  &:active {
    opacity: .7;
  }
}

// Botão com cantos arredondados
.button--round {
  border-radius: 30px;
}

// Botão que ocupa todo o espaço horizontal
.button--block {
  display: block;
  margin: 6px 0;
  width: 100%;
}

// botão com texto em caixa alta
.button--uppercase {
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: bold;
}

// Botão grande
.button--big {
  font-size: 17px;
  line-height: 42px;
  padding: 0 14px;
  height: 44px;
}

// Botão pequeno
.button--small {
  height: 26px;
  padding: 0 8px;
  line-height: 22px;
  border-width: 2px;
  font-size: 13px;
  font-weight: 600;
  text-transform: uppercase;
}

.button--onlyIcon {
  padding: 1px 5px 0;
  margin-right: 6px;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #444
}

// Botão com cor azul
.button--blue {
  border-color: @blue-color;
  color: @blue-color;

  &:not(.button--fill):active {
    background-color: fade(@blue-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@blue-color, 30%);
    z-index: 10;
  }

  // Botão azul com preenchimento
  &.button--fill {
    background-color: @blue-color;
    border-color: transparent;
    color: #FFF;
  }
}

// Botão com cor vermelha
.button--red {
  border-color: @red-color;
  color: @red-color;

  &:not(.button--fill):active {
    background-color: fade(@red-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@red-color, 50%);
    z-index: 10;
  }

  // botão vermelho com preenchimento
  &.button--fill {
    background-color: @red-color;
    border-color: transparent;
    color: #FFF;
  }
}

// Botão com cor amarela
.button--yellow {
  border-color: @yellow-color;
  color: darken(@yellow-color, 40%);

  &:not(.button--fill):active {
    background-color: fade(@yellow-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@yellow-color, 50%);
    z-index: 10;
  }

  // Botão amarelo com preenchimento
  &.button--fill {
    background-color: @yellow-color;
    border-color: transparent;
    color: darken(@yellow-color, 40%);
  }
}

// Botão com cor verde
.button--green {
  border-color: @green-color;
  color: @green-color;

  &:not(.button--fill):active {
    background-color: fade(@green-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@green-color, 50%);
    z-index: 10;
  }

  // Botão verde com preenchimento
  &.button--fill {
    background-color: @green-color;
    border-color: transparent;
    color: #FFF;
  }
}

// Botão com cor cinza
.button--gray {
  border-color: @gray-color;
  color: @gray-color;

  &:not(.button--fill):active {
    background-color: fade(@gray-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@gray-color, 50%);
    z-index: 10;
  }

  // Botão verde com preenchimento
  &.button--fill {
    background-color: @gray-color;
    border-color: transparent;
    color: #FFF;
  }
}

// Botão com cor vermelha
.button--white {
  border-color: @white-color;
  color: @white-color;

  &:not(.button--fill):active {
    background-color: fade(@white-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@white-color, 50%);
    z-index: 10;
  }

  // botão vermelho com preenchimento
  &.button--fill {
    background-color: @white-color;
    border-color: transparent;
    color: #333;
  }
}

// Botão com cor preta
.button--black {
  border-color: @black-color;
  color: @black-color;

  &:not(.button--fill):active {
    background-color: fade(@black-color, 10%);
  }

  &:not(:active):focus {
    box-shadow: 0 0 0 3px fade(@black-color, 50%);
    z-index: 10;
  }

  // botão vermelho com preenchimento
  &.button--fill {
    background-color: @black-color;
    border-color: transparent;
    color: #FFFFFF;
  }
}
