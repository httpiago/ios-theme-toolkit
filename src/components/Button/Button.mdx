---
name: Button
route: /components/Button
menu: Componentes
fullpage: false
---

import { Playground, PropsTable } from 'docz'
import { Button } from '../../index.js'
import '../../../dist/styles.css'

# Button

To use this component, simply import it into any file and pass the [modifiers](#/#props) in the properties.

```jsx
import { Button } from 'ios-theme-toolkit'

export default () => (
  <Button fill color="green">Click me</Button>
)
```

## Examples

<Playground>
  <Button>Button</Button>
  <Button fill>Filled</Button>
  <Button outline>No borders</Button>
  <Button fill round>Rounded</Button>
  <Button size="big">BIG BUTTON</Button>
  <Button size="small">Small</Button>
  <Button fill size="small">Small filled</Button>
  <Button disabled>Disabled</Button>
  <Button href="#">Link button</Button>
  <span className="desc">Only icon:</span>
  <Button onlyIcon><svg width={18} height={18} viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth={2} strokeLinecap="round" strokeLinejoin="round"><path d="M22 17H2a3 3 0 0 0 3-3V9a7 7 0 0 1 14 0v5a3 3 0 0 0 3 3zm-8.27 4a2 2 0 0 1-3.46 0"/></svg></Button>
  <Button onlyIcon fill round color="gray"><svg width={18} height={18} viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth={2} strokeLinecap="round" strokeLinejoin="round"><path d="M22 17H2a3 3 0 0 0 3-3V9a7 7 0 0 1 14 0v5a3 3 0 0 0 3 3zm-8.27 4a2 2 0 0 1-3.46 0"/></svg></Button>
  <br />

  <Button block fill size="big" uppercase>Button that occupies the entire horizontal space</Button>

  <span className="desc">Colorful:</span>
  <Button fill color="red">Attention!</Button>
  <Button fill color="green">Success</Button>
  <Button fill color="yellow">Caution</Button>
  <Button fill color="gray">OK</Button>
  <span style={{ backgroundColor:'black', padding:6, display:'inline-block' }}>
    <Button fill color="white">White</Button>
  </span>
  <Button fill color="black">Black</Button>
  <br />
  <Button color="red">Red</Button>
  <Button color="green">Green</Button>
  <Button color="yellow">Yellow</Button>
  <Button color="gray">Gray</Button>
  <span style={{ backgroundColor:'black', padding:6, display:'inline-block' }}>
    <Button color="white">White</Button>
  </span>
  <Button fill color="black">Black</Button>

  <style>{".react-live-preview .desc { display: block; font-weight: bold; margin-top: 20px; }"}</style>
  <style>{" .react-live-preview .button { margin: 0 3px; } "}</style>
</Playground>

## Props

<PropsTable of={Button} />

All the properties which are not listed above will be transferred to the \<button\> tag, so, It accepts all props which native button support.
