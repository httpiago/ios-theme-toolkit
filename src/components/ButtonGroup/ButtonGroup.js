import React from 'react'
import PropTypes from 'prop-types'
import { mapClassList } from '../../Utils'

/**
 * Component responsible for creating the ButtonGroup.
 * 
 * @since 0.1.0
 * @see https://httpiago.github.io/ios-theme-toolkit/#/components/ButtonGroup
 */
export default function ButtonGroup({ children, round, className: aditionalClasses, ...rest }) {
  
  const prefix = 'button-group'
  const classes = mapClassList({
    [`${prefix}--round`]: round
  })
  
  return (
    <div
      className={`${prefix} ${classes} ${aditionalClasses}`}
      {...rest}
    >{children}</div>
  )
}

ButtonGroup.defaultProps = {  
  round: false,
  className: ''
}

ButtonGroup.propTypes = {
  /** Anything that can be rendered by React. */
  children: PropTypes.node,
  /** Round edges. */
  round: PropTypes.bool,
  /** Aditional classes. */
  className: PropTypes.string,
}
