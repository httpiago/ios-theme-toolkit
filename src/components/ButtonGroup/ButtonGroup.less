@import '../../customizations.less';

// Default styles
.button-group {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin: 20px 0;
}

// Fazer os botões ficarem colados uns nos outros
.button-group .button {
  border-radius: 0;
  border-left-width: 0;
  margin: 0px !important;

  &:first-child {
    border-radius: @base-border-radius 0 0 @base-border-radius;
    border-left-width: 1px;
  }
  &:last-child {
    border-radius: 0 @base-border-radius @base-border-radius 0;
  }
}

// Arredondar os cantos do primeiro e do último botão
.button-group--round .button {
  &:first-child {
    border-radius: 29px 0 0 29px;
  }
  &:last-child {
    border-radius: 0 29px 29px 0;
  }
}
