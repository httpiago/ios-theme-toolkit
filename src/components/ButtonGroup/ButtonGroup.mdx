---
name: ButtonGroup
route: /components/ButtonGroup
menu: Componentes
---

import { Playground, PropsTable } from 'docz'
import { ButtonGroup, Button } from '../../index.js'
import '../../../dist/styles.css'

# Grouped buttons (segmented)

To use this component, simply import it into any file and pass the [modifiers](#/#props) in the properties.

```jsx
import { ButtonGroup, Button } from 'ios-theme-toolkit'

export default () => (
  <ButtonGroup>
    <Button>First</Button>
    <Button fill>Second</Button>
    <Button>Third</Button>
  </ButtonGroup>
)
```

## Examples

<Playground>
  <ButtonGroup>
    <Button>First</Button>
    <Button fill>Second</Button>
    <Button>Third</Button>
    <Button>Fourth</Button>
  </ButtonGroup>

  <ButtonGroup round>
    <Button>First</Button>
    <Button>Second</Button>
    <Button fill>Third</Button>
    <Button>Fourth</Button>
  </ButtonGroup>

  <span className="desc">Vertical buttons:</span>
  <ButtonGroup direction="vertical">
    <Button>First</Button>
    <Button>Second</Button>
    <Button fill>Third</Button>
    <Button>Fourth</Button>
  </ButtonGroup>

  <style>{".react-live-preview .desc { display: block; font-weight: bold; margin-top: 20px }"}</style>
</Playground>

## Props

<PropsTable of={ ButtonGroup } />

All the properties which are not listed above will be transferred to the \<div class="button-group"\> tag.
