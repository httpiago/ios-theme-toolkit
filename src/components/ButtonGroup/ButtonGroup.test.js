/**
 * TEST FILE
 * Please try writing the tests for all component behaviors,
 * manually call the functions within the class, change the properties...
 *
 * @see https://jestjs.io/docs/en/expect
 * @see https://github.com/airbnb/enzyme/tree/master/docs/api
 */
import React from 'react'
import { shallow } from 'enzyme'
import ButtonGroup from './ButtonGroup'
import Button from '../Button/Button'

const COMPONENT = shallow(
  <ButtonGroup className="aditionalClass">
    <Button>Button 1</Button>
    <Button>Button 2</Button>
    <Button>Button 3</Button>
  </ButtonGroup>
)

describe('Test ButtonGroup component', () => {
  it('should render correctly based on last snapshot', () => {
    expect(COMPONENT).toMatchSnapshot()
  })

  it('should update correctly', () => {
    COMPONENT.setProps({ round: true })

    expect(COMPONENT.hasClass('button-group--round')).toBeTruthy()
  })

  it('should has aditional classes', () => {
    expect(COMPONENT.hasClass('aditionalClass')).toBeTruthy()
  })

  it('should render buttons as children correctly', () => {
    expect(COMPONENT.children()).toHaveLength(3)

    // expect(COMPONENT.children().first().type()).toEqual('button')
  })

  it('should pass aditional props to html element', () => {
    COMPONENT.setProps({ 'data-custom-attr': 'yes' })

    expect(COMPONENT.html()).toContain('data-custom-attr="yes"')
  })

})
