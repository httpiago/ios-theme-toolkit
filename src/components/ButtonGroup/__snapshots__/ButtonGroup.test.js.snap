// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonGroup component should render correctly based on last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonGroup
    className="aditionalClass"
    outline={false}
    round={false}
    vertical={false}
  >
    <Button
      block={false}
      className=""
      color="blue"
      disabled={false}
      fill={false}
      htmlType="button"
      onClick={[Function]}
      onlyIcon={false}
      outline={false}
      round={false}
      size="default"
      uppercase={false}
    >
      Button 1
    </Button>
    <Button
      block={false}
      className=""
      color="blue"
      disabled={false}
      fill={false}
      htmlType="button"
      onClick={[Function]}
      onlyIcon={false}
      outline={false}
      round={false}
      size="default"
      uppercase={false}
    >
      Button 2
    </Button>
    <Button
      block={false}
      className=""
      color="blue"
      disabled={false}
      fill={false}
      htmlType="button"
      onClick={[Function]}
      onlyIcon={false}
      outline={false}
      round={false}
      size="default"
      uppercase={false}
    >
      Button 3
    </Button>
  </ButtonGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          block={false}
          className=""
          color="blue"
          disabled={false}
          fill={false}
          htmlType="button"
          onClick={[Function]}
          onlyIcon={false}
          outline={false}
          round={false}
          size="default"
          uppercase={false}
        >
          Button 1
        </Button>,
        <Button
          block={false}
          className=""
          color="blue"
          disabled={false}
          fill={false}
          htmlType="button"
          onClick={[Function]}
          onlyIcon={false}
          outline={false}
          round={false}
          size="default"
          uppercase={false}
        >
          Button 2
        </Button>,
        <Button
          block={false}
          className=""
          color="blue"
          disabled={false}
          fill={false}
          htmlType="button"
          onClick={[Function]}
          onlyIcon={false}
          outline={false}
          round={false}
          size="default"
          uppercase={false}
        >
          Button 3
        </Button>,
      ],
      "className": "button-group button-group--horizontal aditionalClass",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": false,
          "children": "Button 1",
          "className": "",
          "color": "blue",
          "disabled": false,
          "fill": false,
          "htmlType": "button",
          "onClick": [Function],
          "onlyIcon": false,
          "outline": false,
          "round": false,
          "size": "default",
          "uppercase": false,
        },
        "ref": null,
        "rendered": "Button 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": false,
          "children": "Button 2",
          "className": "",
          "color": "blue",
          "disabled": false,
          "fill": false,
          "htmlType": "button",
          "onClick": [Function],
          "onlyIcon": false,
          "outline": false,
          "round": false,
          "size": "default",
          "uppercase": false,
        },
        "ref": null,
        "rendered": "Button 2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": false,
          "children": "Button 3",
          "className": "",
          "color": "blue",
          "disabled": false,
          "fill": false,
          "htmlType": "button",
          "onClick": [Function],
          "onlyIcon": false,
          "outline": false,
          "round": false,
          "size": "default",
          "uppercase": false,
        },
        "ref": null,
        "rendered": "Button 3",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            block={false}
            className=""
            color="blue"
            disabled={false}
            fill={false}
            htmlType="button"
            onClick={[Function]}
            onlyIcon={false}
            outline={false}
            round={false}
            size="default"
            uppercase={false}
          >
            Button 1
          </Button>,
          <Button
            block={false}
            className=""
            color="blue"
            disabled={false}
            fill={false}
            htmlType="button"
            onClick={[Function]}
            onlyIcon={false}
            outline={false}
            round={false}
            size="default"
            uppercase={false}
          >
            Button 2
          </Button>,
          <Button
            block={false}
            className=""
            color="blue"
            disabled={false}
            fill={false}
            htmlType="button"
            onClick={[Function]}
            onlyIcon={false}
            outline={false}
            round={false}
            size="default"
            uppercase={false}
          >
            Button 3
          </Button>,
        ],
        "className": "button-group button-group--horizontal aditionalClass",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": false,
            "children": "Button 1",
            "className": "",
            "color": "blue",
            "disabled": false,
            "fill": false,
            "htmlType": "button",
            "onClick": [Function],
            "onlyIcon": false,
            "outline": false,
            "round": false,
            "size": "default",
            "uppercase": false,
          },
          "ref": null,
          "rendered": "Button 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": false,
            "children": "Button 2",
            "className": "",
            "color": "blue",
            "disabled": false,
            "fill": false,
            "htmlType": "button",
            "onClick": [Function],
            "onlyIcon": false,
            "outline": false,
            "round": false,
            "size": "default",
            "uppercase": false,
          },
          "ref": null,
          "rendered": "Button 2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": false,
            "children": "Button 3",
            "className": "",
            "color": "blue",
            "disabled": false,
            "fill": false,
            "htmlType": "button",
            "onClick": [Function],
            "onlyIcon": false,
            "outline": false,
            "round": false,
            "size": "default",
            "uppercase": false,
          },
          "ref": null,
          "rendered": "Button 3",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
